src/lib.rs

```diff
@@ -116,7 +119,19 @@ mod arch {
             command.arg(perf_output);
         }
 
+        // perf script can take a long time to run. Notify the user that it is running
+        // by using a spinner. Note that if this function exits before calling
+        // spinner.finish(), then the spinner will be completely removed from the terminal.
+        let spinner = ProgressBar::new_spinner().with_prefix("Running perf script");
+        spinner.set_style(
+            ProgressStyle::with_template("{prefix} [{elapsed}]: {spinner:.green}")
+                .unwrap()
+                .tick_strings(&[".", "..", "...", " ..", "  .", "", "✓"]),
```

Comment:
Why deviate from the default style?

Reply:
Let's split this into two parts: the template and the spinner.

In terms of changing the template, I think there are some important deviations from the default:
 1. I don't believe there is a way to show the elapsed time in the default style template. I tried calling `with_elapsed(Duration::from_millis(0))` on the progress bar instantiation using the default style, and I don't see anything. I'm not a power user of indicatif, so maybe I am doing it wrong, but setting the style with the template seemed like the best way to include the elapsed time. **I feel strongly** that including the elapsed time is important since I cannot find a way to implement a true progress bar. If there are better ways to include the elapsed time, we can use those.
 2. The default style does not include a prefix that comes before the spinner. You can include a message which comes after the spinner, but I personally think that looks strange. I am used to seeing something of the form `<what is happening>: <status>`. I **don't** feel as strongly about this one, so if you prefer the message after the spinner that is fine.

Given that, please let me know how you want to proceed with changing the template.


For the spinner itself, we can switch back to the default tick strings. I personally didn't like it in part because the bottom of the spinner doesn't line up with the bottom of the prompt, but any style or color works. Shall I go ahead and switch back to the default spinner?

Reply:
I think I'm fine with the template, was mostly responding to the custom spinner. Let's remove that for now? If you think the default spinner is bad, consider advocating a change upstream (where I'm also a maintainer).

Reply:
> If you think the default spinner is bad, consider advocating a change upstream (where I'm also a maintainer).

I wouldn't say "bad", just the bottom of the spinner didn't line up with the bottom of my prefix text. No big deal, and possibly even specific to how my system is rendering it. Like I said before, I'm not a good person to consult on style decisions anyway!